homeassistant/components/flux_led/light.py

```diff
@@ -300,21 +300,48 @@ def effect(self) -> str | None:
 
     async def _async_turn_on(self, **kwargs: Any) -> None:
         """Turn the specified or all lights on."""
-        if (brightness := kwargs.get(ATTR_BRIGHTNESS)) is None:
-            brightness = self.brightness
-
         if not self.is_on:
             await self._device.async_turn_on()
-            if not kwargs:
-                return
+        if not kwargs:
+            return
+        if effect := kwargs.get(ATTR_EFFECT):
+            await self._async_set_effect(effect)
+            return
```

Comment:
This feels odd to me... if a service call is made that sets an effect and the brightness, the brightness is never applied?

Reply:
This is expected

![Screenshot 2021-11-23 at 3 20 35 PM](https://user-images.githubusercontent.com/663432/143131085-efa633f0-6ff1-4594-969b-3468b29654a2.png)

No brightness control in app either

Reply:
Alright, no flux led for me it is then 😬 😄

Reply:
Sent a few test packets to the device and guessed the brightness byte.

Turns out its actually possible but the app and the library don't support it.

Reply:
https://github.com/home-assistant/core/pull/60247